main.tf

```diff
@@ -59,7 +59,7 @@ resource "aws_s3_bucket_versioning" "default" {
 }
 
 resource "aws_s3_bucket_logging" "default" {
-  count = local.enabled && try(length(var.logging), 0) > 0 ? 1 : 0
+  for_each = toset(local.enabled && length(var.logging) > 0 ? ["enabled"] : [])
```

Comment:
Could you use `moved` to avoid breaking changes when bumping versions?

```
moved {
  from = xxx[0]
  to   = xxx["enabled"]
}
```

Reply:
@taoufik07 
Yes

Reply:
This is a good call. I wasn't thinking this was necessary as the destroy / create lifecycle that would happen in downstream projects that consume this module would likely not have a material impact on the access logging, but I definitely agree this is preferable. 

That said, this is the first I've personally seen usage of a `moved` block in one of our modules, so I am checking with the rest of the maintainer team to see if there are any conventions we've been following surrounding these types of blocks e.g. maybe these need to live in `migrations.tf` or similar.

I'll follow up with another review once I've heard more on that. 

Thanks @taoufik07 + @wadhah101 👍